### DISCUSSION TITLE
Why are backpacks based on bundles rather than chests?

### DISCUSSION THREAD

**Discussion by GoldenFurredFox** [BeansGalaxy/Beans-Backpacks-2#60] [Feedback] [answered]
I ask this with no malice or anything.

I am just curious, since other backpack mods offer a chest like interface. The fact that they behave like bundles does offer a curious opportunity for storage, but it also offers some limitations, such as that if you want to offer 27 slots of storage like a chest does, you'd have to choose between offering only 27 spaces, which is less than a vanilla bundle, or 1728 spaces, which could lead to a game crash if you packed it to the brim with unstackable items and died if no limits on drops were imposed.

I could totally understand keeping the limitation, but also potentially offering the ability to put a Shulker Box there or something. Shulkers are end-game anyways, so it shouldn't prove to be too much of an issue, though it might also be a neat idea to offer a custom backpack using two shulker boxes (or two chests and four shells) to offer a double chest in a backpack, especially if it can then nest other Shulkers.

**Comment by BeansGalaxy** [comment, marked as answer]
No malice taken, backpacks are based on bundles because most modded backpacks have always been just a movable chest. Now since minecraft added shulker boxes, backpack mods just feel like you got a shulker box on clearance and aren't really needed. Bundles are super cool items too, but I am NOT setting up a rabbit farm for an item that can store 64 items. (I've done it before and it sucked) 

As for shulker boxes, I think they're powerful enough and storing more on the player in a backpack feels like storing backpacks in backpacks. I'm trying to add more features that make backpacks feel more like an inventory upgrade versus new shulker boxes like using pick block searches through them, having access to them from the inventory, and picking up items goes into them. lmk if you have any more ideas.

I am curious how the metal/netherite backpacks feel, I recently lowered how many items they hold but I'm thinking it might be too little. I don't want to destroy the "economy" of minecraft and put End Cities out of business but, if you've played with the higher tiers of backpacks how does the size feel?

**Reply by GoldenFurredFox** [reply]
It feels alright, I guess? It's arguably a little bit worse, since upgrading versions deleted any excess items in my pack, but at the end of the day, I could (and did) just use /give to return what I lost.

**Reply by BeansGalaxy** [reply]
Shoot, I tested it and I thought it stopped you from placing items into it until you removed the extra items, must've changed something last minute. I'll double check the code to make sure your items are protected in the future. 

Thanks for letting me know, that's why the mod's still in beta.

**Reply by GoldenFurredFox** [reply]
Yeah, it did stop me from putting anything new in, but it also just ate all the items that were beyond the limit the moment I tried to click any items into the pack. I restored a world backup and tried to remove items instead, and it did save a single item from the very front of that extended area from being voided, but the rest still got voided.
